[ DOC ] setBatchSize update (#3316)
openvinotoolkit/openvino: docs/IE_DG/ShapeInference.md

* **Set a new input shape** with the `InferenceEngine::CNNNetwork::reshape` method.<br>
   The `InferenceEngine::CNNNetwork::reshape` method updates input shapes and propagates them down to the outputs of the model through all intermediate layers. 
   > **NOTES**:
   > - Starting with the 2021.1 release, the Model Optimizer converts topologies keeping shape-calculating sub-graphs by default, which enables correct shape propagation during reshaping.
   > - Older versions of IRs are not guaranteed to reshape successfully. Please regenerate them with the Model Optimizer of the latest version of OpenVINO™.<br>
   > - If an ONNX model does not have a fully defined input shape and the model was imported with the ONNX importer, reshape the model before loading it to the plugin.
* **Set a new batch dimension value** with the `InferenceEngine::CNNNetwork::setBatchSize` method.<br>     
   The meaning of a model batch may vary depending on the model design.

>    Batch dimension assumed to be placed on the 0 index for all inputs of the model.

Reviewer: --> `Batch dimension is usually placed at the 0 index of all inputs of the model. `

   This method does not work for models with a non-zero index batch placement or models with inputs without a batch dimension. 

>    The `InferenceEngine::CNNNetwork::setBatchSize` method transforms the model before new shape propagation to relax hard-coded batch dimension in the model if any.

Reviewer: --> `The method transforms the model before a new shape propagation to relax a hard-coded batch dimension in the model, if any.`

> You can change input shapes of a model multiple times using `InferenceEngine::CNNNetwork::reshape` and `InferenceEngine::CNNNetwork::setBatchSize` methods in any order.

Reviewer: --> 
```To change input shapes multiple times, use the `InferenceEngine::CNNNetwork::reshape` and `InferenceEngine::CNNNetwork::setBatchSize` methods in any order.```

> If a model has hard-coded batch dimension, use `InferenceEngine::CNNNetwork::setBatchSize` first to change the batch and then call `InferenceEngine::CNNNetwork::reshape` to update other dimensions if needed.

Reviewer: --> ```If a model has a hard-coded batch dimension, use `InferenceEngine::CNNNetwork::setBatchSize` first to change the batch, then call `InferenceEngine::CNNNetwork::reshape` to update other dimensions, if needed.```

> Once input shape of the `InferenceEngine::CNNNetwork` is set call `InferenceEngine::Core::LoadNetwork` method to get `InferenceEngine::ExecutableNetwork` object for inference with updated shapes.

Reviewer: --> ```Once the input shape of `InferenceEngine::CNNNetwork` is set, call the `InferenceEngine::Core::LoadNetwork` method to get an `InferenceEngine::ExecutableNetwork` object for inference with updated shapes.```

   
> **NOTES**:
> - Starting with the 2021.1 release, the Model Optimizer converts topologies keeping shape-calculating sub-graphs by default, which enables correct shape propagation during reshaping.
> - Older versions of IRs are not guaranteed to reshape successfully. Please regenerate them with the Model Optimizer of the latest version of OpenVINO™.<br>
> - If an ONNX model does not have a fully defined input shape and the model was imported with the ONNX importer, reshape the model before loading it to the plugin.

* **Set a new batch dimension value** with the `InferenceEngine::CNNNetwork::setBatchSize` method.<br>     
   The meaning of a model batch may vary depending on the model design.
   Batch dimension is usually placed at the 0 index of all inputs of the model.
   This method does not work for models with a non-zero index batch placement or models with inputs without a batch dimension. 
   The method transforms the model before a new shape propagation to relax a hard-coded batch dimension in the model, if any.

> To change input shapes multiple times, use the `InferenceEngine::CNNNetwork::reshape` and `InferenceEngine::CNNNetwork::setBatchSize` methods in any order.

Reviewer: ```suggestion
You can change input shapes multiple times using the `InferenceEngine::CNNNetwork::reshape` and `InferenceEngine::CNNNetwork::setBatchSize` methods in any order.
```

Changing the model input shape may result in intermediate operations shape collision.

Examples of such operations:

> - <a href="_docs_ops_shape_Reshape_1.html">`Reshape` operation</a> with a hard-coded output shape value

Reviewer: These should work:
```
- [`Reshape` operation](../ops/shape/Reshape_1.md) with a hard-coded output shape value
- [`MatMul` operation](../ops/matrix/MatMul_1.md) with the `Const` second input cannot be resized by spatial dimensions due to operation semantics
```

> - <a href="_docs_ops_shape_Reshape_1.html">`Reshape` operation</a> with a hard-coded output shape value
> - <a href="_docs_ops_matrix_MatMul_1.html">`MatMul` operation</a> with the `Const` second input cannot be resized by spatial dimensions due to operation semantics

Author: ```suggestion
- [`Reshape` operation](../ops/shape/Reshape_1.md) with a hard-coded output shape value
- [`MatMul` operation](../ops/matrix/MatMul_1.md) with the `Const` second input cannot be resized by spatial dimensions due to operation semantics
```

> - <a href="_docs_ops_shape_Reshape_1.html">`Reshape` operation</a> with a hard-coded output shape value
> - <a href="_docs_ops_matrix_MatMul_1.html">`MatMul` operation</a> with the `Const` second input cannot be resized by spatial dimensions due to operation semantics

Author: ```suggestion
- [`Reshape` operation](../ops/shape/Reshape_1.md) with a hard-coded output shape value
- [`MatMul` operation](../ops/matrix/MatMul_1.md) with the `Const` second input cannot be resized by spatial dimensions due to operation semantics
```
